src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,117 @@
+     *
+     * @param input the user input
+     * @return a <code>Command</code> equivalent to the user input
+     * @throws DukeException if the input format is invalid
+     */
+    public static Command parse(String input) throws DukeException {
+        String[] args = input.trim().split("\\s+");
+        Command command;
+        switch (args[0]) {
+        case "bye":
+            command = new ExitCommand();
+            break;
+        case "list":
+            command = new ListCommand();
+            break;
+        case "find":
+            command = new FindCommand(
+                    Arrays.stream(args).skip(1).reduce("", (x, y) -> x + " " + y).trim());
+            break;
+        case "mark":
+            try {
+                command = new MarkCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "unmark":
+            try {
+                command = new UnMarkCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "delete":
+            try {
+                command = new DeleteCommand(Integer.parseInt(args[1]) - 1);
+            } catch (IndexOutOfBoundsException exception) {
+                throw new DukeException("OOPS!!! No task index is specified :(");
+            } catch (NumberFormatException exception) {
+                throw new DukeException("OOPS!!! You didn't give a valid index :(");
+            }
+            break;
+        case "todo":
+            command = new AddCommand(new ToDo(
+                    Arrays.stream(args).skip(1).reduce("", (x, y) -> x + " " + y).trim(),
+                    false));
+            break;
+        case "event":
+            try {
+                String description = Arrays.stream(args)
+                        .takeWhile(x -> !x.equals("/at")).skip(1)
+                        .reduce("", (x, y) -> x + " " + y).trim();
+                String[] timeArgument = Arrays.stream(args)
+                        .dropWhile(x -> !x.equals("/at")).toArray(String[]::new);
+                if (timeArgument.length <= 1) {
+                    throw new DukeException("OOPS!!! There is no /at argument for event :(");
+                }
+                command = new AddCommand(
+                        new Event(description, false, LocalDate.parse(timeArgument[1])));
+            } catch (java.time.format.DateTimeParseException exception) {

```

Comment:
Great catch of the exception here. Might it be better (and shorter) if `DateTimeParseException` is imported so that we do not need to include `java.time.format` at the front all the time?